### PR TITLE
Viking Template and added PDS3 capabilities in isisimport

### PR DESCRIPTION
**Pull request by amystamile-usgs**
## Description

Viking template with added PDS3 importing capabilities to isisimport. Still needs testing to confirm output of viking image matches output from vik2isis.

## Motivation and Context

PDS4 Sprint Nov 2021

## How Has This Been Tested?

Will add tests in separate PR.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/PvlToJSON/PvlToJSON.cpp [hunk 1]
```diff
@@ -349,4 +351,24 @@ namespace Isis {
...
+    Pvl pvl(pvlFile);
+    QFile file(pvlFile);
+
+    if (!file.open(QIODevice::ReadOnly)) {
+      QString message = QString("Failed to open file for PVL Input: [%1]").arg(pvlFile);
+      throw IException(IException::Io, message, _FILEINFO_);
+    }
+    file.close();
+
+    return pvlToJSON(pvl);
```
Comment (acpaquette) [inline isis/src/base/objs/PvlToJSON/PvlToJSON.cpp]: Looking over the Pvl constructor from a QString, it takes care of the majority of the necessary tests on the file. I think we can just wrap line 363 in a try catch block then just rethrow the error.

File: isis/src/base/apps/isisimport/isisimport.cpp [hunk 2]
```diff
@@ -22,33 +23,41 @@ namespace Isis {
...
+    catch(...) {
+      try {
+        // try to convert pvl to json
+        jsonData = pvlToJSON(inputFileName.toString());
+      }
+      catch(...) {
+        QString msg = "Unable to process import image. Please confirm image is in PDS3 or PDS4 format";
+        throw IException(IException::User, msg, _FILEINFO_);
+      }
     }
```
Comment (acpaquette) [inline isis/src/base/apps/isisimport/isisimport.cpp:50]: I want to say there is a better way to do this but I am not sure. Ideally we could enumerate our drivers, current that is only pvlToJson and xmlToJson, the just loop over and try each of them. ISIS has typically done this with factories but I think we could potentially just operate using a list of functions

**Review by acpaquette — COMMENTED** [review]
Overall looks good! One suggested change and a one potential change to discuss

File: isis/src/base/apps/isisimport/isisimport.cpp (hunk 2 above)
Comment (amystamile-usgs) [reply isis/src/base/apps/isisimport/isisimport.cpp:50]: I agree. I feel there is a better way as well, but was unsure how to keep a tracker for whether or not the format was an XML. This is due to the fact that if it's XML , we need to add the OriginalXmlLabel to the cube.

Comment (scsides) [reply isis/src/base/apps/isisimport/isisimport.cpp:50]: Agreed, let's add this as a todo/think about

**Review by scsides — APPROVED** [review]

**Review by scsides — APPROVED** [review]
Unless someone has a reason, go ahead and merge this. I want to start removing duplicates after we make a decision

**Pull request merged**